base/applications/taskmgr/perfdata.c

```diff
@@ -221,22 +221,13 @@ void PerfDataRefresh(void)
     }
 
     /* Get handle information
-     * We don't know how much data there is so just keep
-     * increasing the buffer size until the call succeeds
+     * Number of handles is enough, no need for data array.
      */
-    BufferSize = 0;
-    do
-    {
-        BufferSize += 0x10000;
-        SysHandleInfoData = (LPBYTE)HeapAlloc(GetProcessHeap(), 0, BufferSize);
-
-        status = NtQuerySystemInformation(SystemHandleInformation, SysHandleInfoData, BufferSize, &ulSize);
-
-        if (status == STATUS_INFO_LENGTH_MISMATCH) {
-            HeapFree(GetProcessHeap(), 0, SysHandleInfoData);
-        }
-
-    } while (status == STATUS_INFO_LENGTH_MISMATCH);
+    status = NtQuerySystemInformation(SystemHandleInformation, &SysHandleInfoData, sizeof(SysHandleInfoData), NULL);
+    // On unexpected error, reuse previous value.
+    // STATUS_SUCCESS (0-1 handle) should never happen.
```

Comment:
Consistency with other comments in this file.
```suggestion
    /* On unexpected error, reuse previous value.
     * STATUS_SUCCESS (0-1 handle) should never happen.
     */
```

Reply:
Done.